src/components/NamespaceList/useNamespaceActions.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { getUserDataFromLocalStorage } from '../../auth/utils';
+import { k8sCreateResource } from '../../k8s/k8s-fetch';
+import { RoleBindingModel, SelfSubjectAccessReviewModel } from '../../models';
+import { Action } from '../../shared/components/action-menu/types';
+import { NamespaceKind } from '../../types';
+import { SelfSubjectAccessReviewKind } from '../../types/rbac';
+import { useModalLauncher } from '../modal/ModalProvider';
+import { createManageVisibilityModalLauncher } from './ManageVisibilityModalLauncher';
+
+export const useNamespaceActions = (
+  namespace: NamespaceKind,
+): [Action[], boolean, (isOpen: boolean) => void] => {
+  const showModal = useModalLauncher();
+  const [isChecking, setChecking] = React.useState(false);
+  const [canManage, setCanManage] = React.useState(false);
+  const [checked, setChecked] = React.useState(false);
+  const [permissionError, setPermissionError] = React.useState<string | null>(null);
+
+  const checkPermissions = React.useCallback(async () => {
+    setChecking(true);
+    setPermissionError(null);
+    const user = getUserDataFromLocalStorage();
+    const resourceAttributes = {
+      group: RoleBindingModel.apiGroup,
+      resource: RoleBindingModel.plural,
+      namespace: namespace.metadata.name,
+    };
+    try {
+      const [createResponse, deleteResponse] = await Promise.all([
```

Comment:
I did a quick check in our codebase and I found a util func called `checkReviewAccesses` - it seems we can use it here to simplify the code:

```tsx
      const hasPermission = await checkReviewAccesses(
        [
          { model: RoleBindingModel, verb: 'create' },
          { model: RoleBindingModel, verb: 'delete' },
        ],
        namespace.metadata.name,
      );
```